### PR TITLE
[Navigation Item] Use Icon component with `untrusted` instead of `dangerouslySetInnerHTML`

### PR DESCRIPTION
**Pull request by theodoretan**
### WHY are these changes introduced?

Resolves https://github.com/Shopify/polaris-react/issues/885

Inlining user uploaded svgs is potentially dangerous as scripts can be executed if they are part of the svg. In a previous PR, I updated the `Icon` component to take an `untrusted` prop that will tell the component to render the contents in an `img` tag instead of an svg.

### WHAT is this pull request doing?

This PR changes how Item works by using the Icon component for svg strings instead of using `dangerouslySetInnerHTML`. To make this work, I updated the `recolor-icon` mixin that set's the fill for the icon. I choose to make the `$filter-color` the last param to keep the current functionality for icons.

~⚠️ **Note**: Online Store's icon doesn't work with this solution so if we are okay with this, we'd have to update the Online Store sales channel.~
Edit: Fixed!
image

<details>
<summary>I 🎩ed with the current sales channels a user can install from the channels modal:</summary>

image

</details>

<details>
<summary>Showing the `focus`, `hovered`, and base states:</summary>

ec7b42ee-dd2f-4e2f-819f-d75e92adec91

</details>

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

Copy-paste this code in <code>playground/Playground.tsx</code>:

```jsx
import * as React from 'react';
import {Page, Navigation} from '../src';

interface State {}

const onlinestore = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path fill="currentColor" d="M19 6c0 1.657-1.343 3-3 3s-3-1.343-3-3c0 1.657-1.343 3-3 3S7 7.657 7 6c0 1.657-1.343 3-3 3S1 7.657 1 6V5l2-4h14l2 4v1z"></path><path d="M16 8c-1.103 0-2-.897-2-2h4c0 1.103-.897 2-2 2zm0 6H4v-4c1.193 0 2.267-.525 3-1.357C7.733 9.475 8.807 10 10 10s2.267-.525 3-1.357c.733.832 1.807 1.357 3 1.357v4zm-3.28 4H7.28c.358-.702.537-1.434.628-2h4.184c.09.566.27 1.298.627 2zM12 6c0 1.103-.897 2-2 2s-2-.897-2-2h4zM2 6h4c0 1.103-.897 2-2 2s-2-.897-2-2zm1.618-4h12.764l1 2H2.618l1-2zm16.277 2.553l-2-4C17.725.213 17.38 0 17 0H3c-.38 0-.725.214-.895.553l-2 4C.035 4.69 0 4.845 0 5v1c0 1.474.81 2.75 2 3.444V15c0 .552.447 1 1 1h2.87c-.156.747-.507 1.7-1.317 2.105-.415.208-.633.673-.527 1.125.108.45.51.77.974.77h10c.464 0 .866-.32.974-.77.106-.452-.112-.917-.527-1.125-.8-.4-1.153-1.356-1.313-2.105H17c.553 0 1-.448 1-1V9.444c1.19-.694 2-1.97 2-3.444V5c0-.155-.036-.31-.105-.447z"></path></svg>`;

const amazon = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><title>amazon-icon-16px</title><path d="M14.23,13.18a13.73,13.73,0,0,1-6,1.46A14.15,14.15,0,0,1,.25,12.16c-.2-.14-.34.1-.18.28A10.82,10.82,0,0,0,8,16a10.32,10.32,0,0,0,6.49-2.34C14.77,13.41,14.52,13,14.23,13.18Zm1.71-1a3.41,3.41,0,0,0-2.89.25c-.16.1-.13.24,0,.23.58-.06,1.86-.21,2.08.07s-.25,1.4-.47,1.9c-.06.15.08.21.22.1A2.84,2.84,0,0,0,15.94,12.22ZM9.39,4.57a16.46,16.46,0,0,0-4.23.73,3.57,3.57,0,0,0-2.4,3.43A3.09,3.09,0,0,0,6.13,12a4.6,4.6,0,0,0,3.71-1.53,5.25,5.25,0,0,0,1.29,1.4.49.49,0,0,0,.53,0h0c.45-.37,1.26-1,1.72-1.4.18-.14.15-.37,0-.56a2.78,2.78,0,0,1-.85-1.95V4.66c0-1.39.1-2.66-1-3.62A5.34,5.34,0,0,0,8.17,0C6,0,3.64.75,3.13,3.22a.37.37,0,0,0,.34.44l2.18.22A.46.46,0,0,0,6,3.49,1.7,1.7,0,0,1,7.85,2.23a1.55,1.55,0,0,1,1.26.55A2.54,2.54,0,0,1,9.39,4.3ZM9,8.94a1.84,1.84,0,0,1-1.56,1A1.35,1.35,0,0,1,6,8.37C6,6.58,7.76,6.26,9.39,6.26v.46A3.89,3.89,0,0,1,9,8.94Z"></path></svg>`;

const pos = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 18 20"><path d="M8.717 9.46l.77-2.26s-.52-.29-1.57-.29c-2.71 0-4.06 1.8-4.06 3.66 0 2.21 2.22 2.27 2.22 3.61 0 .33-.23.77-.8.77-.87 0-1.9-.88-1.9-.88l-.53 1.73s1.01 1.21 2.97 1.21c1.64 0 2.85-1.22 2.85-3.12 0-2.42-2.7-2.81-2.7-3.85 0-.18.06-.93 1.26-.93.82 0 1.49.35 1.49.35zm-2.34-5.62c-.51.14-.88.57-.94 1.09-.04.42.2.85.7.81s.9-.51 1-.97c.11-.49-.18-1.04-.76-.93zm1.35-2.7c-.32 0-.72.55-.84 1.36l1.6-.39c-.19-.64-.53-.97-.76-.97zm1.93.9c.32.09.62.29.81.58l1.62 2.13c.16.23.196.458.18.78l-.23 4.84-.33 6.68-.1 1.96c0 .13-.01.26-.03.38-.09.49-.51.67-.97.59l-8.65-1.62c-.51-.09-1.07-.15-1.57-.29-.6-.17-.34-1.1-.28-1.55l.36-2.78.82-6.23c.03-.21.05-.42.08-.63.05-.28.17-.55.34-.78l1.68-2.39c.18-.28.47-.48.8-.55l1.21-.3.3-.07c.09-1.58.94-2.79 2.04-2.79.9 0 1.65.86 1.92 2.04zM12 2l3.875 2.616c.168.157.276.37.31.598l1.77 13.008c.05.195 0 .4-.135.55-.13.15-.33.226-.527.2l-3.61-.37.066-1.36.33-6.864.25-5.208c.01-.182-.05-.36-.164-.503L12 2z"></path></svg>`;

const ebay = `<svg xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="0 0 1000 1000"><path d="M935.3,908.6c-0.7-16.6-3.1-33.1-4.7-49.6c-3.2-31.9-6.4-63.8-9.6-95.6c-2.9-28.6-5.7-57.2-8.5-85.7c-3.2-31.7-6.4-63.5-9.6-95.2c-3.2-31.7-6.3-63.5-9.5-95.2c-3.2-32.2-6.5-64.4-9.7-96.5c-1.4-13.7-1.7-27.5-4.1-41c-7-39.6-40.4-67.2-80.6-67.4c-25.4-0.1-50.8,0-76.2,0c-1.6,0-3.2,0-5.2,0c0-17.3,0.5-34.1-0.1-50.8c-0.5-14.2-1.3-28.4-3.6-42.4c-14-85.7-84.1-157.6-169.6-174.4c-10.2-2-20.6-3.1-30.9-4.6c-9.1,0-18.1,0-27.2,0c-1.2,0.2-2.3,0.6-3.5,0.7c-49.1,4-92.3,22.4-128.7,55.5c-42.6,38.7-66.7,86.9-71,144.4c-1.6,22.1-0.7,44.4-1,66.6c0,1.6,0,3.2,0,5c-2.3,0-4,0-5.6,0c-26.2,0-52.3-0.2-78.5,0.1c-37.2,0.5-71,28.4-77.3,65.1c-3.5,20.4-4.4,41.1-6.5,61.7c-2.9,28.6-5.7,57.1-8.5,85.7c-3.2,32-6.4,64.1-9.6,96.1c-3.1,31.4-6.3,62.9-9.4,94.3c-3.5,34.7-7,69.5-10.5,104.2c-3.2,32.3-6.5,64.7-9.7,97c-0.9,9.3-2.5,18.8-1.8,28c3.1,43,38.3,75.5,81.5,75.5c235.8,0.1,471.5,0,707.3,0c11,0,21.6-2.1,31.8-6.3C915.4,971.3,936.6,940.2,935.3,908.6z M336.7,224.2c0.9-76.6,58.7-144.2,134.3-157.1c88.8-15.1,171.1,40.9,189.2,129c1.5,7.5,2.7,15.3,2.8,22.9c0.4,20.8,0.1,41.7,0.1,63.1c-108.8,0-217.3,0-326.2,0c-0.1-1.2-0.3-2.3-0.3-3.5C336.6,260.4,336.5,242.3,336.7,224.2z M836.6,934.6c-3.9,0.6-7.8,0.9-11.7,0.9c-216.6,0-433.1,0-649.7,0c-29.4,0-51.9-18.8-55.6-47.8c-1.1-8.7,0.6-17.7,1.6-26.6c3.7-34.3,7.6-68.5,11.4-102.7c4.1-36.8,8.2-73.6,12.3-110.4c3.5-31.2,6.9-62.5,10.4-93.7c3.5-31.2,7-62.5,10.4-93.7c2.7-24.5,5.6-48.9,7.7-73.5c2.4-27.7,25.4-50,53.1-50.4c17.5-0.2,35.1-0.1,52.6-0.1c0.9,0,1.8,0.1,3.1,0.2c0,20,0.1,39.6-0.1,59c0,1.6-1.8,3.6-3.3,4.6c-33.7,23.4-30.9,74.1,5.2,93.2c25.4,13.4,56.8,4.8,71.9-19.6c15.1-24.6,8.5-56.7-15.5-73.1c-2.9-2-3.8-4-3.8-7.3c0.2-18.7,0.1-37.5,0.1-56.6c108.8,0,217.5,0,326.4,0c0.1,1.3,0.2,2.6,0.2,3.9c0,17.8-0.1,35.7,0.1,53.5c0,2.9-1,4.5-3.4,6.2c-20,13.6-28.6,38.1-21.5,61c7,22.7,28,38.3,51.7,38.4c23.8,0.1,44.9-15.2,52.1-37.8c7.3-22.9-1-47.6-20.8-61.3c-2.8-1.9-3.9-3.8-3.9-7.3c0.2-17.4,0.1-34.8,0.1-52.2c0-1.5,0-3,0-4.8c8.3,0,16.1-0.1,23.9,0c13.1,0.2,26.4-0.6,39.4,1c24.6,3,43.5,24.6,45.6,49.3c1.6,19,3.8,37.9,5.9,56.8c3.1,28.7,6.4,57.4,9.6,86.1c3.5,31.4,7,62.8,10.5,94.2c3.2,28.5,6.3,57.1,9.5,85.6c3.5,31.4,7,62.8,10.5,94.2c2.8,25.2,5.9,50.4,8.4,75.7C883.3,904.6,862.4,930.3,836.6,934.6z"></path></svg>`;

const instagram = `<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 16 16" version="1.1"><title>Group</title><g><g><path d="M7.99749206,0 C5.82549206,0 5.55314286,0.00920634921 4.70012698,0.0481269841 C3.84888889,0.086952381 3.26752381,0.22215873 2.7588254,0.419873016 C2.23292063,0.624222222 1.78692063,0.89768254 1.34228571,1.34228571 C0.89768254,1.78692063 0.624222222,2.23292063 0.419873016,2.7588254 C0.22215873,3.26752381 0.086952381,3.84888889 0.0481269841,4.70012698 C0.00920634921,5.55314286 0,5.82549206 0,7.99749206 C0,10.1694603 0.00920634921,10.4418095 0.0481269841,11.2948254 C0.086952381,12.1460635 0.22215873,12.7274286 0.419873016,13.236127 C0.624222222,13.7620317 0.89768254,14.2080317 1.34228571,14.6526667 C1.78692063,15.0972698 2.23292063,15.3707302 2.7588254,15.5751111 C3.26752381,15.7727937 3.84888889,15.908 4.70012698,15.9468254 C5.55314286,15.985746 5.82549206,15.9949524 7.99749206,15.9949524 C10.1694603,15.9949524 10.4418095,15.985746 11.2948254,15.9468254 C12.1460635,15.908 12.7274286,15.7727937 13.236127,15.5751111 C13.7620317,15.3707302 14.2080317,15.0972698 14.6526667,14.6526667 C15.0972698,14.2080317 15.3707302,13.7620317 15.5751111,13.236127 C15.7727937,12.7274286 15.908,12.1460635 15.9468254,11.2948254 C15.985746,10.4418095 15.9949524,10.1694603 15.9949524,7.99749206 C15.9949524,5.82549206 15.985746,5.55314286 15.9468254,4.70012698 C15.908,3.84888889 15.7727937,3.26752381 15.5751111,2.7588254 C15.3707302,2.23292063 15.0972698,1.78692063 14.6526667,1.34228571 C14.2080317,0.89768254 13.7620317,0.624222222 13.236127,0.419873016 C12.7274286,0.22215873 12.1460635,0.086952381 11.2948254,0.0481269841 C10.4418095,0.00920634921 10.1694603,0 7.99749206,0 Z M7.99749206,1.44098413 C10.1328889,1.44098413 10.3858413,1.44914286 11.2291429,1.48761905 C12.0088889,1.5231746 12.4323492,1.65346032 12.7141587,1.76298413 C13.0874603,1.90806349 13.353873,2.08136508 13.6337143,2.3612381 C13.9135873,2.64107937 14.0868889,2.90749206 14.2319683,3.28079365 C14.3414921,3.56260317 14.4717778,3.98606349 14.5073333,4.76580952 C14.5458095,5.60911111 14.5539683,5.86206349 14.5539683,7.99749206 C14.5539683,10.1328889 14.5458095,10.3858413 14.5073333,11.2291429 C14.4717778,12.0088889 14.3414921,12.4323492 14.2319683,12.7141587 C14.0868889,13.0874603 13.9135873,13.353873 13.6337143,13.6337143 C13.353873,13.9135873 13.0874603,14.0868889 12.7141587,14.2319683 C12.4323492,14.3414921 12.0088889,14.4717778 11.2291429,14.5073333 C10.3859683,14.5458095 10.1330476,14.5539683 7.99749206,14.5539683 C5.86190476,14.5539683 5.60901587,14.5458095 4.76580952,14.5073333 C3.98606349,14.4717778 3.56260317,14.3414921 3.28079365,14.2319683 C2.90749206,14.0868889 2.64107937,13.9135873 2.3612381,13.6337143 C2.08139683,13.353873 1.90806349,13.0874603 1.76298413,12.7141587 C1.65346032,12.4323492 1.5231746,12.0088889 1.48761905,11.2291429 C1.44914286,10.3858413 1.44098413,10.1328889 1.44098413,7.99749206 C1.44098413,5.86206349 1.44914286,5.60911111 1.48761905,4.76580952 C1.5231746,3.98606349 1.65346032,3.56260317 1.76298413,3.28079365 C1.90806349,2.90749206 2.08136508,2.64107937 2.3612381,2.3612381 C2.64107937,2.08136508 2.90749206,1.90806349 3.28079365,1.76298413 C3.56260317,1.65346032 3.98606349,1.5231746 4.76580952,1.48761905 C5.60911111,1.44914286 5.86206349,1.44098413 7.99749206,1.44098413 Z"></path><path d="M7.99749206,10.6633016 C6.5251746,10.6633016 5.33165079,9.46977778 5.33165079,7.99749206 C5.33165079,6.5251746 6.5251746,5.33165079 7.99749206,5.33165079 C9.46977778,5.33165079 10.6633016,6.5251746 10.6633016,7.99749206 C10.6633016,9.46977778 9.46977778,10.6633016 7.99749206,10.6633016 Z M7.99749206,3.89066667 C5.72933333,3.89066667 3.89066667,5.72933333 3.89066667,7.99749206 C3.89066667,10.265619 5.72933333,12.1042857 7.99749206,12.1042857 C10.265619,12.1042857 12.1042857,10.265619 12.1042857,7.99749206 C12.1042857,5.72933333 10.265619,3.89066667 7.99749206,3.89066667 Z"></path><path d="M13.226254,3.7284127 C13.226254,4.25844444 12.7965714,4.68809524 12.2665397,4.68809524 C11.7365397,4.68809524 11.3068571,4.25844444 11.3068571,3.7284127 C11.3068571,3.19838095 11.7365397,2.76869841 12.2665397,2.76869841 C12.7965714,2.76869841 13.226254,3.19838095 13.226254,3.7284127"></path></g></g></svg>`;

const facebook = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path d="M14.566 0H1.434C.946 0 0 .947 0 1.434v13.13c0 .488.946 1.434 1.434 1.434H8.27V9.924H6.07V7.728h2.204v-1.69c0-2.065 1.564-3.292 3.408-3.292.883 0 1.348.095 1.57.124v2.09h-1.565c-1.003 0-1.19.498-1.19 1.198v1.57h2.784l-.364 2.197h-2.42V16h4.07c.487 0 1.434-.947 1.434-1.434V1.434C16 .947 15.054 0 14.566 0z"></path></svg>`;

const messenger = `<svg xmlns="http://www.w3.org/2000/svg" version="1.1" viewBox="0 0 16 16"><path d="M8,0 C3.58171429,0 0,3.31639506 0,7.40740741 C0,9.73851852 1.16332919,11.8178272 2.98136646,13.1756543 L2.98136646,16 L5.70529193,14.505037 C6.43224845,14.7062222 7.20238509,14.8148148 8,14.8148148 C12.4182857,14.8148148 16,11.4984198 16,7.40740741 C16,3.31639506 12.4182857,0 8,0 L8,0 Z M8.78947368,10 L6.8277512,7.65957447 L3,10 L7.21052632,5 L9.22009569,7.34042553 L13,5 L8.78947368,10 L8.78947368,10 Z"></path></svg>`;

const buy = `<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 20 20" version="1.1"><title>icon-buy_button</title><g><g><g><g><path d="M11.5200002,19.8880004 L17.5040004,18.5920004 C17.5040004,18.5920004 15.3440003,3.98400009 15.3280003,3.88800008 C15.3120003,3.79200008 15.2320003,3.72800008 15.1520003,3.72800008 C15.0720003,3.72800008 13.5520003,3.61600008 13.5520003,3.61600008 C13.5520003,3.61600008 12.4960003,2.56000006 12.3680003,2.44800005 C12.3360003,2.41600005 12.3040003,2.40000005 12.2720003,2.38400005 L11.5200002,19.8880004 Z"></path><path d="M12.0160003,2.35200005 C12.0000003,2.35200005 11.9680003,2.36800005 11.9520003,2.36800005 C11.9360003,2.36800005 11.7120003,2.43200005 11.3600002,2.54400006 C11.0080002,1.52000003 10.3840002,0.576000012 9.2800002,0.576000012 C9.2480002,0.576000012 9.2160002,0.576000012 9.1840002,0.576000012 C8.86400019,0.176000004 8.48000018,0 8.14400018,0 C5.56800012,0 4.33600009,3.21600007 3.95200009,4.84800011 C2.96000006,5.15200011 2.24000005,5.37600012 2.16000005,5.40800012 C1.60000003,5.58400012 1.58400003,5.60000012 1.52000003,6.12800013 C1.45600003,6.51200014 0,17.7920004 0,17.7920004 L11.2480002,19.9040004 L12.0160003,2.35200005 Z M9.0880002,3.13600007 C9.0880002,3.16800007 9.0880002,3.21600007 9.0880002,3.24800007 C8.46400018,3.44000007 7.77600017,3.64800008 7.10400015,3.85600008 C7.48800016,2.38400005 8.20800018,1.66400004 8.83200019,1.39200003 C8.99200019,1.80800004 9.0880002,2.36800005 9.0880002,3.13600007 Z M8.06400017,0.688000015 C8.17600018,0.688000015 8.28800018,0.720000016 8.40000018,0.800000017 C7.58400016,1.18400003 6.68800014,2.16000005 6.32000014,4.11200009 C5.77600013,4.28800009 5.24800011,4.4480001 4.7520001,4.5920001 C5.18400011,3.10400007 6.22400013,0.688000015 8.06400017,0.688000015 Z M8.51200018,9.3760002 C8.51200018,9.3760002 7.84000017,9.0240002 7.04000015,9.0240002 C5.84000013,9.0240002 5.79200013,9.77600021 5.79200013,9.96800022 C5.79200013,10.9920002 8.48000018,11.3920002 8.48000018,13.8080003 C8.48000018,15.7120003 7.28000016,16.9280004 5.64800012,16.9280004 C3.69600008,16.9280004 2.70400006,15.7120003 2.70400006,15.7120003 L3.23200007,13.9840003 C3.23200007,13.9840003 4.25600009,14.8640003 5.12000011,14.8640003 C5.68000012,14.8640003 5.92000013,14.4160003 5.92000013,14.0960003 C5.92000013,12.7520003 3.71200008,12.6880003 3.71200008,10.4800002 C3.71200008,8.62400019 5.04000011,6.81600015 7.74400017,6.81600015 C8.78400019,6.81600015 9.2960002,7.12000015 9.2960002,7.12000015 L8.51200018,9.3760002 Z M9.53600021,1.28000003 C10.1120002,1.34400003 10.4800002,2.00000004 10.7200002,2.73600006 C10.4320002,2.83200006 10.1120002,2.92800006 9.76000021,3.04000007 C9.76000021,2.97600006 9.76000021,2.91200006 9.76000021,2.83200006 C9.76000021,2.20800005 9.68000021,1.69600004 9.53600021,1.28000003 Z"></path></g></g></g></g></svg>`;

const wholesale = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 14" version="1.1"><title>Monotone</title><path d="M9.6,12.8 C9.1592,12.8 8.8,12.4408 8.8,12 C8.8,11.5592 9.1592,11.2 9.6,11.2 C10.0408,11.2 10.4,11.5592 10.4,12 C10.4,12.4408 10.0408,12.8 9.6,12.8 L9.6,12.8 Z M2.4,12.8 C1.9592,12.8 1.6,12.4408 1.6,12 C1.6,11.5592 1.9592,11.2 2.4,11.2 C2.8408,11.2 3.2,11.5592 3.2,12 C3.2,12.4408 2.8408,12.8 2.4,12.8 L2.4,12.8 Z M2.4,6.4 L4,6.4 L4,4.8 L2.4,4.8 L2.4,6.4 Z M8.3056,3.2 L9.9056,6.4 L5.6,6.4 L5.6,3.2 L8.3056,3.2 Z M7.3472,11.2 L4.6528,11.2 C4.3208,10.2704 3.4416,9.6 2.4,9.6 C2.1176,9.6 1.8512,9.6576 1.6,9.7472 L1.6,8 L10.4,8 L10.4,9.7472 C10.1488,9.6576 9.8824,9.6 9.6,9.6 C8.5584,9.6 7.6792,10.2704 7.3472,11.2 L7.3472,11.2 Z M15.2,9.6 L13.6,9.6 L13.6,2.4 C13.6,1.9584 13.2416,1.6 12.8,1.6 C12.3584,1.6 12,1.9584 12,2.4 L12,7.2 C12,7.1912 11.9952,7.184 11.9952,7.176 C11.9936,7.1224 11.976,7.0712 11.9632,7.0176 C11.9504,6.9664 11.944,6.9136 11.9224,6.8664 C11.9184,6.8584 11.9192,6.8504 11.916,6.8424 L9.516,2.0424 C9.38,1.7712 9.1032,1.6 8.8,1.6 L4.8,1.6 C4.3584,1.6 4,1.9584 4,2.4 L4,3.2 L1.6,3.2 C1.1584,3.2 0.8,3.5584 0.8,4 L0.8,6.4 C0.3584,6.4 0,6.7584 0,7.2 L0,12 C0,13.3232 1.0768,14.4 2.4,14.4 C3.4416,14.4 4.3208,13.7296 4.6528,12.8 L7.348,12.8 C7.6792,13.7296 8.5584,14.4 9.6,14.4 C10.9232,14.4 12,13.3232 12,12 L12,10.4 C12,10.8416 12.3584,11.2 12.8,11.2 L15.2,11.2 C15.6416,11.2 16,10.8416 16,10.4 C16,9.9584 15.6416,9.6 15.2,9.6 L15.2,9.6 Z"></path></svg>`;

const houzz = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" enable-background="new 0 0 16 16"><path d="M8 16l4-2v-6l-4 2zM4 12l4-2-4-2zM8 6l4 2v-5zM8 0l-4 3v5l4-2z"></path></svg>`;

const buzz = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 38 38"><title>Viral Arrow</title><path d="M19 38C8.5 38 0 29.5 0 19S8.5 0 19 0s19 8.5 19 19-8.5 19-19 19zm9.9-19.4l-1.3-9-8.4 3.4 3.3 1.9-3.1 5.4-5.4-3-5.4 9.3 3 1.7 3.6-6.3 5.4 3.1 4.9-8.5 3.4 2z"></path></svg>`;

const kik = `<svg viewBox="0 0 224 224" xmlns="http://www.w3.org/2000/svg"><title>Kik_dot</title><path d="M188.999 108C202.256 108 213 118.746 213 132c0 13.255-10.744 24-24.001 24C175.743 156 165 145.255 165 132c0-13.254 10.743-24 23.999-24zm-75.451 19.735l39.141 53.015c7.272 9.85 5.124 23.738-4.786 30.93-9.774 7.093-23.436 5-30.638-4.694l-37.458-50.422-11.082 10.28v27.295C68.722 206.213 58.934 216 46.861 216 34.788 216 25 206.212 25 194.138l.001-167.172c0-12.42 10.302-22.413 22.794-21.95 11.877.44 21.114 10.556 21.138 22.44l.17 81.647 41.184-40.335c8.464-8.289 21.95-8.455 30.614-.376 9.223 8.6 9.42 23.153.432 31.998l-27.785 27.345z" fill-rule="evenodd"></path></svg>`;

export default class Playground extends React.Component<{}, State> {
  render() {
    return (
      <Page title="Playground">
        <Navigation location="/">
          <Navigation.Section
            items={[
              {
                url: '/path/to/place',
                label: 'Online Store',
                iconBody: onlinestore,
              },
              {
                url: '/path/to/place',
                label: 'Amazon',
                iconBody: amazon,
                badge: '15',
              },
              {
                url: '/path/to/place',
                label: 'Point of Sale',
                iconBody: pos,
              },
              {
                url: '/path/to/place',
                label: 'eBay',
                iconBody: ebay,
              },
              {
                url: '/path/to/place',
                label: 'Instagram',
                iconBody: instagram,
                badge: '15',
              },
              {
                url: '/path/to/place',
                label: 'Facebook',
                iconBody: facebook,
              },
              {
                url: '/path/to/place',
                label: 'Buy Button',
                iconBody: buy,
              },
              {
                url: '/path/to/place',
                label: 'Wholesale',
                iconBody: wholesale,
                badge: '15',
              },
              {
                url: '/path/to/place',
                label: 'Houzz',
                iconBody: houzz,
              },
              {
                url: '/path/to/place',
                label: 'Messenger',
                iconBody: messenger,
                badge: '15',
              },
              {
                url: '/path/to/place',
                label: 'BuzzFeed',
                iconBody: buzz,
              },
              {
                url: '/path/to/place',
                label: 'Kik',
                iconBody: kik,
              },
            ]}
          />
        </Navigation>
      </Page>
    );
  }
}
```

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)

### CODE REVIEW

File: src/components/Navigation/Navigation.scss [hunk 1]
```diff
@@ -99,7 +99,11 @@ $disabled-fade: 0.6;
 }
 
 .SecondaryAction {
-  @include recolor-icon(color('ink', 'lightest'), color('white'));
+  @include recolor-icon(
+    color('ink', 'lightest'),
+    color('white'),
+    filter('ink', 'lightest')
```
Comment (amrocha) [inline src/components/Navigation/Navigation.scss]: Why did you add these filters?

File: src/styles/shared/icons.scss [hunk 2]
```diff
@@ -2,8 +2,16 @@
...
-@mixin recolor-icon($main-color, $secondary-color: null) {
-  fill: $main-color;
+@mixin recolor-icon($fill-color, $secondary-color: null, $filter-color: null) {
+  svg {
+    fill: $fill-color;
+  }
+
+  @if $filter-color != null {
+    img {
+      filter: $filter-color;
```
Comment (amrocha) [inline src/styles/shared/icons.scss]: Looking at the docs for `filter` (https://developer.mozilla.org/en-US/docs/Web/CSS/filter) it doesn't look like you can just pass in a color string. How does this work?

Comment (theodoretan) [reply src/styles/shared/icons.scss]: There's a map of filter values that match with the polaris colors from `polaris-tokens`: https://github.com/Shopify/polaris-tokens/blob/master/dist/color-filters.json

In a previous PR, I added the filter scss function that allows us to use grab the values using the same naming convention.

Comment (amrocha) [reply src/styles/shared/icons.scss]: Oh I see, I didn't realize the values were coming from that file. This code makes sense then, and I'd rather use the Icon component to render icons than spans so this is good IMO

**Comment by amrocha** [comment]
>**Note**: Online Store's icon doesn't work with this solution so if we are okay with this, we'd have to update the Online Store sales channel.

So to be clear, the Online Store's icon in web is rendered differently now because of this, right? @HYPD do you know if there's a way of fixing this?

File: src/components/Navigation/Navigation.scss (hunk 1 above)
Comment (ry5n) [reply src/components/Navigation/Navigation.scss]: Looks like this uses the revised `recolor-icon` mixin to handle all cases here: the first 2 params handle the case where the icon is an inline SVG, the other when the icon is an `img`.

File: src/styles/shared/icons.scss (hunk 2 above)
Comment (theodoretan) [reply src/styles/shared/icons.scss]: The `$filter-color` variable is the value that comes back from the `filter` function. The filter function handles the validation of the input to ensure it is the correct color. So this variable would be something like:

 `brightness(0) saturate(100%) invert(28%) sepia(83%) saturate(3919%) hue-rotate(168deg) brightness(93%) contrast(101%)`

**Comment by theodoretan** [comment]
> I haven’t tophatted but this makes sense to me. Based on the images in the PR description, which I’m assuming are a before/after, the Online Store icon now looks correct—the filled-in roof is not how it should look if I recall

Sorry for the confusion 😅. The image above is in opposite order, so the one of the left is the after image. I've updated it to have labels now.

With the solution in this PR, the Online Store icon will have a filled-in roof, unless we update it. The Partners Dashboard does the same rendering technique as here (using the Polaris Icon component) so we will be able ensure the icon is correct on upload. This should be done **before** this PR is shipped.

File: src/styles/shared/icons.scss (hunk 2 above)
Comment (ry5n) [reply src/styles/shared/icons.scss]: Derp, sorry I missed that! You’re absolutely right 👍

**Comment by ry5n** [comment]
I did some quick tophatting in Firefox and Safari (Mac) and Safari (iOS) and all looks good. Would be good to check other browsers if not already.

Then like you say, just the online store icon to update and this can go in.

**Comment by theodoretan** [comment]
Update! With the help of Online Store folks we've updated the sales channel icon so now all sales channels will work with this implementation

image

**Comment by theodoretan** [comment]
@amrocha I updated the unreleased changelog, let me know if you think this is good to ship

File: UNRELEASED.md [hunk 3]
```diff
@@ -8,6 +8,8 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Breaking changes
```
Comment (amrocha) [inline UNRELEASED.md]: I don't think this is a breaking change, there's no action that consumers need to take, right? Unless I'm misunderstanding something I think this is more of an `Enhancement`

**Comment by amrocha** [comment]
One more comment, the code itself looks good

File: UNRELEASED.md (hunk 3 above)
Comment (theodoretan) [reply UNRELEASED.md]: Thanks, I updated this to be under `Enhancement`
